ably/options.go

```diff
@@ -14,36 +15,72 @@ const (
 	ProtocolMsgPack = "msgpack"
 )
 
+var errInvalidKey = errors.New("invalid key format")
+
+var DefaultOptions = &ClientOptions{
+	RestEndpoint:     "https://rest.ably.io",
+	RealtimeEndpoint: "wss://realtime.ably.io:443",
+	Protocol:         ProtocolJSON, // TODO: make it ProtocolMsgPack
+}
+
 type ClientOptions struct {
-	RealtimeEndpoint string
 	RestEndpoint     string
-	Key              string
+	RealtimeEndpoint string
 	Token            string
 	Secret           string
 	ClientID         string
-	Protocol         string
+	Protocol         string // either ProtocolJSON or ProtocolMsgPack
 	UseTokenAuth     bool
-	NoTLS            bool
 
 	HTTPClient *http.Client
 }
 
-func (p *ClientOptions) Prepare() {
-	if p.Key != "" {
-		p.parseKey()
+var (
+	restTLS       = strings.NewReplacer("http://", "https://")
+	restNoTLS     = strings.NewReplacer("https://", "http://")
+	realtimeTLS   = strings.NewReplacer("ws://", "wss://", ":443", ":80")
+	realtimeNoTLS = strings.NewReplacer("wss://", "ws://", ":80", ":443")
+)
+
+func (opts *ClientOptions) SetTLS(enabled bool) {
+	if enabled {
+		opts.RestEndpoint = restTLS.Replace(opts.RestEndpoint)
+		opts.RealtimeEndpoint = realtimeTLS.Replace(opts.RealtimeEndpoint)
+	} else {
+		opts.RestEndpoint = restNoTLS.Replace(opts.RestEndpoint)
+		opts.RealtimeEndpoint = realtimeNoTLS.Replace(opts.RealtimeEndpoint)
 	}
 }
 
-func (p *ClientOptions) parseKey() {
-	keyParts := strings.Split(p.Key, ":")
+func (opts *ClientOptions) SetKey(key string) error {
+	s := strings.Split(key, ":")
+	if len(s) != 2 {
+		return errInvalidKey
+	}
+	opts.Token = s[0]
+	opts.Secret = s[1]
+	return nil
+}
+
+func (opts *ClientOptions) Key() string {
+	if opts.Token == "" || opts.Secret == "" {
+		return ""
+	}
+	return opts.Token + ":" + opts.Secret
```

Comment:
The token and key are two completely separate things?  `Token` is an expiring token that is created by a [requestToken](http://docs.ably.io/rest-api/#requesttoken) request, whereas a Key is made up of a key name and key secret, although mostly users will not be separating the two and simply passing a `key` option to the library when instanced.